### PR TITLE
fix(cache): change how the game folder is retrieved

### PR DESCRIPTION
**Pull request by xiaoxiao921**
Use GetModuleFileNameW instead of std::filesystem::current_path()

### CODE REVIEW

File: src/services/gta_data/yim_fipackfile.cpp [hunk 1]
```diff
@@ -84,16 +84,58 @@ namespace big
 		return output;
 	}
 
+	static std::filesystem::path get_game_folder_path()
+	{
+		std::wstring game_module_path(MAX_PATH, '\0');
+
+		if (GetModuleFileNameW(nullptr, game_module_path.data(), game_module_path.size()))
```
Comment (gir489returns) [inline src/services/gta_data/yim_fipackfile.cpp:91]: When I used to manually map my cheat, I had problems with using the WinAPI to retrieve information about GTA's process. Has this been tested with MMing? I don't know if the project only supports LLA injection or not, but that might be something to consider.

Comment (xiaoxiao921) [reply src/services/gta_data/yim_fipackfile.cpp:91]: I can't even MM with current nightly (without this PR merged in) so ¯\_(ツ)_/¯

Comment (Yimura) [reply src/services/gta_data/yim_fipackfile.cpp:91]: Was the user having problems injecting with manual mapping?

Comment (xiaoxiao921) [reply src/services/gta_data/yim_fipackfile.cpp:91]: No, i'm just suspecting std::filesystem::current_path being able to return bogus result on some systems, so i'm just removing one extra layer of potential non controlled errors

**Review by Yimura — APPROVED** [review]
Is this ready to merge or should we wait for more user input before merging?

**Comment by xiaoxiao921** [comment]
You can merge, i'm confident this is much safer than having to rely on current_path

**Pull request merged**